src/shared/components/CSVExportButton.tsx

```diff
@@ -1,72 +1,90 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import {connect, ConnectedProps} from 'react-redux'
 
 // Components
 import {Button, ComponentStatus, IconFont} from '@influxdata/clockface'
 
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+// Selectors and Actions
+import {getActiveQuery} from 'src/timeMachine/selectors'
+import {runDownloadQuery} from 'src/timeMachine/actions/queries'
 
 // Types
 import {AppState} from 'src/types'
 
-interface StateProps {
-  files: string[] | null
+// Worker
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import downloadWorker from 'worker-plugin/loader!../workers/downloadHelper'
+
+type Props = ConnectedProps<typeof connector>
+
+interface State {
+  isEnabled: boolean
 }
 
-class CSVExportButton extends PureComponent<StateProps, {}> {
+class CSVExportButton extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {isEnabled: false}
+
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker.register(downloadWorker).then(
+        () => this.setState({isEnabled: true}),
+        function (err) {
+          console.error('ServiceWorker registration failed: ', err)
```

Comment:
so I wasn't sure for this one. Since it doesn't appear as an error to the user; instead, the user will simply not have the option to download (the `this.state.isEnabled` = false).

The purpose of the console.error, is in case we have a user complain that they don't have the download button. Then we can ask them to check their console, and give a reason why.

Reply:
Thanks for the explanation. It's a non-actionable error, so there's not really a good reason to show the user a toast. Makes perfect sense.

In this case, I'm starting to lean more and more on the idea that we should send this error to Honeybadger, rather than asking the user to look at their console. The error is less ephemeral when it's in Honeybadger, so we can reference it later. It's kind of amateurish to ask someone paying us money to use our product and get support to open up their console and tell us what went wrong. It's also possible that they had the error, reported it, and it's gone now and they don't know or don't want to recreate it.